test/io.rb

```diff
@@ -140,6 +140,13 @@ def assert_nothing_raised(*exp)
   end
 end
 
+assert "IO#read(n) with n > IO::BUF_SIZE" do
+  r,w = IO.pipe
+  n = IO::BUF_SIZE+1
+  w.write 'a'*n
```

Comment:
(Just a comment)

This will block on the systems where the buffer size of pipe is equal to or smaller than `BUF_SIZE` (= 4096).  I found it is 65536 on most OSs (NetBSD/OpenBSD/macOS/Linux(>=2.6.11)) and 8192 on FreeBSD.